### PR TITLE
Fail2Ban Filters and RKHunter

### PR DESCRIPTION
**Pull request by ChiefGyk**
I have added some Fail2Ban filters in my own config also pulling ideas from prior posts here on Git Hub which have been referenced in my commits. I also added [Blocklist.de ](blocklist.de)lists to be automatically loaded daily via crontab and add reported attackers automatically added to IPTables. This was worked well in my own mail server and dropped reports of attackers tremendously. 
I modified the jail.local to add jails, report to blocklist.de (you will have to add an account to blocklist.de that's why it's commented out) but I also like it to have reports sent to my business and personal emails but not quite sure how to automate pulling the admin email into the server names so I commented that out.
I know there were a few things added that were possibly mentioned in a few other issues. This is really my first actual time using Git and Github for code changes after being a basic GNU/Linux guy for 10 years. So I am sure I have a few issues that can be resolved.
I am pretty sure Fail2Ban can be beefed up quite a bit more than what I have offered as well

### CODE REVIEW

**Comment by ChiefGyk** [comment]
Upon testing I am finding it results in one false positive so far from lwp-request I am troubleshooting it right now seems to be related to not having a PKGMNGR=DPKG value being set

**Comment by yodax** [comment]
Hi, thanks for taking the time to create a pull request and trying to improve mailinabox 😄 

First off, I'm not the maintainer. I do have a couple of suggestions. There is a lot going on in this PR, usually it is better to focus on a single change with a clear purpose. That will make it easier for someone to review the PR and will increase your chances of getting it merged. This will also help with one part of your change not being quite ready, while another part is complete. The complete part could be merged.

I did my best to review the individual parts.

Regarding the fail2ban changes this (I believe) boils down to two changes. Nginx bots and blocklist.de. I'm okay with the bots change.

On [this](https://github.com/mail-in-a-box/mailinabox/pull/870/files#diff-2bd502fa5b471085fbc2698fb3f025d3R17) line there is a commented out section about reporting to blocklist.de. (which you mention in your PR) Since you need an account to do this, we should probably remove these lines. It clutters the config. If a user wants to configure this they can do that via a jail.local file after we have move the jails to jail.d in a different PR.

On [this](https://github.com/mail-in-a-box/mailinabox/pull/870/files#diff-2bd502fa5b471085fbc2698fb3f025d3R69) you add a commented out own cloud jail. This is configuration clutter. Also I'm pretty sure that doesn't work without the changes in this https://github.com/mail-in-a-box/mailinabox/pull/868 Maybe you changed that manually? 

[This](https://github.com/ChiefGyk/mailinabox/blob/cb35e6dd9601dd9d74384faedce4b577bd6f4202/conf/fail2ban/miab-owncloud.conf) is called miab-owncloud but contains the management deamon rules from a different PR (which requires code changes to the daemon to work). But more importantly I think something went wrong with the creation of the file. It contains:

```
ignoreregex =
alon@box:/etc/fail2ban$ cat filter.d/miab-owncloud.conf 
```

The same thing happened with miab-munin.conf.

Regarding the rootkit hunter. I'm not sure if that is something I would want. If someone can hack your server they can remove this tool. But I'll gladly defer to someone with more expertise on the matter. I do have some comments on the setup. (As stated before I think this should be a separate PR)

On [this](https://github.com/mail-in-a-box/mailinabox/pull/870/files#diff-cdb065f2ced88cbecf2383758a5da402R4) the config mentions that you can create a local configuration. I think that would be preferable for us to do and only check that one in. That will make it clear which settings are being overwritten. 

With [this](https://github.com/mail-in-a-box/mailinabox/pull/870/files#diff-cdb065f2ced88cbecf2383758a5da402R92) line as it currently is for a user to receive a notification they would need to have set the root alias. So it might be better to configure the recipient to be admin@$DOMAIN. Should be done in the setup.

Where you download the file ([here](https://github.com/mail-in-a-box/mailinabox/pull/870/files#diff-a849b34223db47d855298d35dfda8042R127) you call wget directly. There is a function (see own cloud setup for instance) that downloads a file and verifies its hash. That way we know that the file hasn't been modified. As a side note, we have had many issues with downloads failing from sourceforge. If we proceed with this change, it should probably be mirrored.

[This](https://github.com/mail-in-a-box/mailinabox/pull/870/files#diff-a849b34223db47d855298d35dfda8042R330) should just be configured in line. Downloading a script to be run daily on any miab from github user content might not be a good idea. :smile: Though it seems that if you edit a gist the old one is still downloaded because of the id. 

Again thanks for all the work, I do appreciate it.

**Comment by ChiefGyk** [comment]
A lot of these were changes I have pushed from my own setup, and am still playing with Github and learning as I go (this is my first time using GitHub or really doing any public code contributions). I am working to clean it up as I go, but I am sort of figuring a way for myself to deploy my own version, while simultaneously sharing my fixes. 
Some of these fixes I did late last night, and I had literally fallen asleep on my laptop before I could clean it up some more. I will correct it in a bit

**Comment by ChiefGyk** [comment]
The clutter is due to a commented out ruleset I brought over from my own OwnCloud 9 setup and was testing, it's purely there for my own notes

**Comment by ChiefGyk** [comment]
Apologies for multiple comments, I just need to point out I am also using this as a learning project for me to shake off years of rust being complacent in IT. So I am going to apply some of the fixes

**Comment by ChiefGyk** [comment]
I uploaded some of the fixes, now I am looking into the RKHunter/wget issues

File: conf/fail2ban/jail.local [hunk 1]
```diff
@@ -14,30 +14,30 @@ action = %(action_mwl)s
 [ssh]
 maxretry = 7
 bantime = 3600
-# action = sendmail-whois-lines[name=ssh, dest="<USER>@<DOMAIN>.<TLD>,<USER2>@<DOMAIN2>.<TLD2>,fail2ban@blocklist.de", sender=fail2ban@box.<SERVER>.<TLD>, sendername="Fail2Ban"]
+action = sendmail-whois-lines[name=ssh, dest="admin@$DOMAIN", sender=fail2ban@$DOMAIN, sendername="Fail2Ban"]
```
Comment (yodax) [inline conf/fail2ban/jail.local]: Is the intent on sending an email on every block? That can be achieved via a setting in the jail.local. This now no longer sends to blocklist.de. I don't think this is necessary.

**Comment by yodax** [comment]
> Apologies for multiple comments, I just need to point out I am also using this as a learning project for me to shake off years of rust being complacent in IT. So I am going to apply some of the fixes.

No problem. I'm trying to help. 

It might be worth waiting on what @JoshData thinks about the changes in general. Otherwise you might be doing a lot of work.

**Comment by JoshData** [comment]
It's too much for me to even begin to review. For the things @ChiefGyk created issues for, let's move those parts of this discussion there.

**Comment by ChiefGyk** [comment]
@yodax I moved the fail2ban-sync script to a local file now I am looking at a secure way for RKHunter

**Comment by yodax** [comment]
@chiefgyk could you look into splitting this up? 

Do you know how branches work in git? You seem to be doing this from your master branch. If you need help I can point you in the right direction.

**Comment by ChiefGyk** [comment]
@yodax I am very unfamiliar with git other than the basic add, commit, pull, and push so I would very much appreciate the help so I may better contribute.

**Comment by ChiefGyk** [comment]
I didn't want to muck it more before learning to split changes, so I created a separate repo for the Rkhunter installer here and the various fixes https://github.com/ChiefGyk/rkhunter_install. I'll merge it when I can learn the branches a bit more, this has all been a wonderful learning experience thus far.

**Pull request closed without merging**